utils/zpool_creation/create_zpool.yml

```diff
@@ -0,0 +1,48 @@
+- block:  
+
+   - name: Create a daemonset with privileged access to create zpool on each node
+     shell: >
+       kubectl create -f /utils/zpool_creation/privilege_ds.yml
+     args: 
+       executable: /bin/bash
+    
+   - name: Confirm that the ds pods are running on all nodes
+     shell: >
+       kubectl get pod -l app=zfs-ds
+       --no-headers -o custom-columns=:status.phase | sort | uniq
+     args: 
+       executable: /bin/bash
+     register: result
+     until: "result.stdout == 'Running'"
+     delay: 5
+     retries: 20
+
+   - name: List the daemonset pods 
+     shell: >
+       kubectl get pods -l app=zfs-ds --no-headers -o custom-columns=:.metadata.name
+     args:
+       executable: /bin/bash
+     register: zfs_ds_pods
```

Comment:
So, pools will be created on all the nodes?

Reply:
yes